yarn-project/pxe/src/note_processor/produce_note_dao.ts

```diff
@@ -89,10 +92,53 @@ export async function produceNoteDaos(
           txHash,
           noteHashes,
           dataStartIndexForTx,
+          unencryptedLogs,
         );
       }
+    } else if ((e as any).message.includes('failed to solve blackbox function: embedded_curve_add')) {
+      // TODO(#8769): This branch is a temporary partial notes delivery solution that should be eventually replaced.
+      // This error occurs when we are dealing with a partial note and is thrown when calling
+      // `note.compute_note_hash()` in `compute_note_hash_and_optionally_a_nullifier` function. It occurs with
+      // partial notes because in the partial flow we receive a note log of a note that is missing some fields
+      // here and then we try to compute the note hash with MSM while some of the fields are zeroed out.
+      for (const functionLogs of unencryptedLogs.functionLogs) {
+        for (const log of functionLogs.logs) {
+          const { data } = log;
+          // It is the expectation that partial notes will have the corresponding unencrypted log be multiple
+          // of Fr.SIZE_IN_BYTES as the partial fields should be simply concatenated.
+          if (data.length % Fr.SIZE_IN_BYTES === 0) {
+            const partialFields = [];
+            for (let i = 0; i < data.length; i += Fr.SIZE_IN_BYTES) {
+              const chunk = data.subarray(i, i + Fr.SIZE_IN_BYTES);
+              partialFields.push(Fr.fromBuffer(chunk));
+            }
+
+            // We concatenate the partial fields with the rest of the fields of the note and we try to produce
+            // the note dao again
+            const payloadWithPartialFields = L1NotePayload.fromBuffer(payload.toBuffer());
+            payloadWithPartialFields.note.items.push(...partialFields);
+
+            ({ incomingNote, incomingDeferredNote } = await produceNoteDaos(
```

Comment:
I was quite happy with how this was looking until I got to the recursive call :sweat_smile: This will definitely need improvements.

Reply:
Agree, but would deal with this once we know whether we'll need the inter-transaction subscription model.